python/pyarrow/_csv.pyx

```diff
@@ -585,6 +605,26 @@ cdef class ParseOptions(_Weakrefable):
         except TypeError:
             return False
 
+    def __repr__(self):
+        return (f"<pyarrow.csv.ParseOptions("
+                f"delimiter={self.delimiter!r}, "
+                f"quote_char={self.quote_char!r}, "
+                f"double_quote={self.double_quote}, "
+                f"escape_char={self.escape_char!r}, "
+                f"newlines_in_values={self.newlines_in_values}, "
+                f"ignore_empty_lines={self.ignore_empty_lines}, "
+                f"invalid_row_handler={self.invalid_row_handler})>")
+
+    def __str__(self):
+        return (f"ParseOptions("
+                f"delimiter={self.delimiter!r}, "
+                f"quote_char={self.quote_char!r}, "
+                f"double_quote={self.double_quote}, "
+                f"escape_char={self.escape_char!r}, "
+                f"newlines_in_values={self.newlines_in_values}, "
+                f"ignore_empty_lines={self.ignore_empty_lines}, "
+                f"invalid_row_handler={self.invalid_row_handler})")
```

Comment:
Thanks for adding this! One thing that I am thinking now is if `invalid_row_handler` and also `timestamp_parsers` could be more readable if not `None`:

```python
timestamp_parsers=[<pyarrow._csv._ISO8601 object at 0x10cd360d0>, '%Y-%m-%d']
```

```python
invalid_row_handler=<pyarrow.tests.test_csv.InvalidRowHandler object at 0x11e828050>
```

Not a blocker, of course, but maybe there's a cleaner way to represent these in the output - not sure off the top of my head though!